assembly/index.ts

```diff
@@ -11,100 +9,95 @@ export function pixelmatch(
 	height: i32,
 
 	// Can't use interfaces to make an options object here: https://docs.assemblyscript.org/basics/limitations#oop
-	threshold: f32,     	// matching threshold (0 to 1); smaller is more sensitive
-	includeAA: bool,    	// whether to skip anti-aliasing detection
-	alpha: f32,				// opacity of original image in diff ouput
-	aaR: f32, 				// r color of anti-aliased pixels in diff output
-	aaG: f32,				// g color of anti-aliased pixels in diff output
-	aaB: f32,				// b color of anti-aliased pixels in diff output
-	diffR: f32,				// r color of different pixels in diff output
-	diffG: f32,				// g color of different pixels in diff output
-	diffB: f32				// b color of different pixels in diff output
+	threshold: f64,   // matching threshold (0 to 1); smaller is more sensitive
+	includeAA: bool,  // whether to skip anti-aliasing detection
+	alpha: f64,       // opacity of original image in diff ouput
+	aaR: f64,         // r color of anti-aliased pixels in diff output
+	aaG: f64,         // g color of anti-aliased pixels in diff output
+	aaB: f64,         // b color of anti-aliased pixels in diff output
+	diffR: f64,       // r color of different pixels in diff output
+	diffG: f64,       // g color of different pixels in diff output
+	diffB: f64        // b color of different pixels in diff output
 ): i32 {
 
 	// No Errors: https://docs.assemblyscript.org/basics/limitations#exceptions
 
 	// Image sizes do not match
 	if (img1.length !== img2.length || (output !== null && output.length !== img1.length)) {
-		return -1; 
+		return -1;
 	}
 
 	// Image data size does not match width/height.
 	if (img1.length !== width * height * 4) {
 		return -2;
 	}
 
+  let img1Ptr = img1.dataStart;
+  let img2Ptr = img2.dataStart;
+  let outputPtr = output.dataStart;
+
 	threshold = isNaN(threshold) ? 0.1 : threshold;
 	includeAA = includeAA || false;
-	alpha = isNaN(alpha) ? 0.1 : alpha;
+	alpha     = isNaN(alpha) ? 0.1 : alpha;
```

Comment:
This might be better as `select(0.1, alpha, isNaN(alpha))`.

Same for line 39 if valid.

Reply:
binaryen should optimize this to select